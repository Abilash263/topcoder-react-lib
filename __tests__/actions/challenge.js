import { actions } from '../../src';

jest.mock('../../src/services/challenges');

test('challenge.fetchChallengeInit', () => {
  expect(actions.challenge.getDetailsInit(123456789)).toMatchSnapshot();
});

test('challenge.fetchSubmissionsInit', () => {
  expect(actions.challenge.getSubmissionsInit(123456789)).toMatchSnapshot();
});


test('challenge.getDetailsDone', () => {
  expect(actions.challenge.getDetailsDone(123456789)).toMatchSnapshot();
});

test('challenge.fetchSubmissionsDone', () => {
  expect(actions.challenge.getSubmissionsDone(123456789, {})).toMatchSnapshot();
});

test('challenge.getActiveChallengesCountInit', () => {
  expect(actions.challenge.getActiveChallengesCountInit()).toMatchSnapshot();
});

test('challenge.getActiveChallengesCountDone', async () => {
  expect(actions.challenge.getActiveChallengesCountDone('handle', 'tokenV3')).toMatchSnapshot();
});
