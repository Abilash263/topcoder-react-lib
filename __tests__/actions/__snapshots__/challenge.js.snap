// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`challenge.fetchChallengeInit 1`] = `
Object {
  "payload": "123456789",
  "type": "CHALLENGE/GET_DETAILS_INIT",
}
`;

exports[`challenge.fetchSubmissionsDone 1`] = `
Object {
  "payload": Promise {},
  "type": "CHALLENGE/GET_SUBMISSIONS_DONE",
}
`;

exports[`challenge.fetchSubmissionsInit 1`] = `
Object {
  "payload": "123456789",
  "type": "CHALLENGE/GET_SUBMISSIONS_INIT",
}
`;

exports[`challenge.getActiveChallengesCountDone 1`] = `
Object {
  "payload": Promise {},
  "type": "CHALLENGE/GET_ACTIVE_CHALLENGES_COUNT_DONE",
}
`;

exports[`challenge.getActiveChallengesCountInit 1`] = `
Object {
  "type": "CHALLENGE/GET_ACTIVE_CHALLENGES_COUNT_INIT",
}
`;

exports[`challenge.getDetailsDone 1`] = `
Object {
  "payload": Promise {},
  "type": "CHALLENGE/GET_DETAILS_DONE",
}
`;
